### PR TITLE
Await unpublish before re-publishing on signal-reconnect

### PR DESCRIPTION
**Pull request by lukasIO**

### CODE REVIEW

**Review by boks1971 — APPROVED** [review]

File: src/room/Room.ts [hunk 1]
```diff
@@ -1131,40 +1131,7 @@ class Room extends (EventEmitter as new () => TypedEmitter<RoomEventCallbacks>)
...
-      this.localParticipant.trackPublications.forEach((pub) => {
-        if (pub.track) {
-          localPubs.push(pub);
-        }
-      });
-
-      await Promise.all(
-        localPubs.map(async (pub) => {
-          const track = pub.track!;
-          this.localParticipant.unpublishTrack(track, false);
```
Comment (lukasIO) [inline src/room/Room.ts:1144]: this was missing an `await`. FWIW the `republishAllTracks` method had the await already, so opted to just use that method instead.

Comment (davidzhao) [reply src/room/Room.ts:1144]: I think previously we didn't automatically republish screenshare/audio tracks.. I'm not too sure why.. I think it makes sense not to make that differentiation though.

**Review by davidzhao — APPROVED** [review]
nice fix!

File: src/room/Room.ts (hunk 1 above)
Comment (lukasIO) [reply src/room/Room.ts:1144]: AFAIK we tried to re-publish them but we didn't try to `restart` them as that's not really possible in the browser

Comment (lukasIO) [reply src/room/Room.ts:1144]: we should probably add that check to the participant method, too. good call :)

**Pull request merged**